lang_java/parsing/parser_java.mly

```diff
@@ -416,6 +416,12 @@ bound: ref_type_and_list { $1 }
 /*(*1 Expressions *)*/
 /*(*************************************************************************)*/
 
+typed_metavar:
+ | LP type_ IDENTIFIER RP { (match $2 with
```

Comment:
FWIW I'd probably put the first statement in an auxiliary function,

e.g.
```ocaml
let fail_if_metavar_on_type: type_ -> () =
  | TClass (((typ, _), _)::_) when typ.[0] = '$' -> failwith ...
  | _ -> ()
```

Then you can just call
`fail_if_metavar_on_type $2;`
here.